src/libponyc/codegen/genfun.c

```diff
@@ -295,6 +321,9 @@ static bool genfun_fun(compile_t* c, reach_type_t* t, reach_method_t* m)
   codegen_startfun(c, m->func, m->di_file, m->di_method);
   name_params(c, t, m, params, m->func);
 
+  if(m->func == t->final_fn)
+    call_embed_finalisers(c, t, LLVMGetParam(codegen_fun(c), 0));
```

Comment:
What do you think about using `gen_this` instead of `LLVMGetParam(..., 0)`?  Since we already have that abstraction available, it seems prudent to try to use it wherever possible, and avoid depending on the order of params here.